src/location.rs

```diff
@@ -0,0 +1,43 @@
+
+/// Represents the source location (row, column) of this element in the original Ion text.
+///
+/// The source location metadata is primarily intended for error reporting and debugging purposes,
+/// helping applications provide meaningful feedback to users about the source of issues.
+///
+/// # Returns
+/// * `Some((row, column))` - Position where this element was found in the source text
+/// * `None` - Location information is not available
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct SourceLocation {
+    location: Option<(usize, usize)>
+}
```

Comment:
Any particular reason you chose to implement `SourceLocation` with an `Option` inside of it instead of using `Option<SourceLocation>` for method return types? Do you anticipate having other source location information in this struct at some point (perhaps `offset`) that would make it possible to have source location without necessarily having a row/column?

Reply:
I think I just meant to keep the multiple usage of `Option<(usize, usize)>` concise with `SourceLocation`. It would keep API cleaner than wrapping everything in `Option<SourceLocation>`. But I havent particularly thought about adding an extension with `offset`. Also, I think if we do we have an offset then we should be able to calculate `row` and `column` based on that.